src/include/migraphx/op/unary_not.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_UNARY_NOT_HPP
+#define MIGRAPHX_GUARD_OPERATORS_UNARY_NOT_HPP
+
+#include <migraphx/op/unary.hpp>
+#include <migraphx/operation.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/config.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct unary_not : unary<unary_not>
+{
```

Comment:
Even though the class has to be `unary_not` so it won't conflict with C++ keywords, maybe we should still have the operator name as `not`, which could be done by adding a `name` method to the class(ie `std::string name() const { return "not"; }`.

Reply:
> @scxiao , how do we implement an operator for GPU? Is it /src/targets/gpu ? Can you provide me a pointer from an existing op?

Please refer to the logical_and operator.